core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BoundedRangeFileInputStream.java

```diff
@@ -62,12 +62,15 @@ public BoundedRangeFileInputStream(FSDataInputStream in, long offset, long lengt
 
   @Override
   public int available() throws IOException {
-    int avail = in.available();
-    if (pos + avail > end) {
-      avail = (int) (end - pos);
-    }
+    final FSDataInputStream inLocal = in;
+    synchronized (inLocal) {
+      int avail = in.available();
```

Comment:
We don't follow the paradigm prescribed by the interface. If used in a parallel environment the block region may all be different. The underlying stream itself will likely be nowhere near your sought position and therefore availability means nothing to the instance that is being called; however, if we used the length provided by the Constructor at least we have some semblance of what availability means for a bounded range input stream. Anyone making a decision based on availability could have a completely incorrect meaning because seek always occurs within a read. This will be further compounded due to threading.